ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/precompiles/privacy/OnChainPrivacyPrecompiledContract.java

```diff
@@ -189,6 +205,20 @@ public Bytes compute(final Bytes input, final MessageFrame messageFrame) {
     return result.getOutput();
   }
 
+  private void sendParticipantRemovedEvent(final PrivateTransaction privateTransaction) {
+    if (privateTransaction.isGroupRemovalTransaction()) {
+      // get first participant parameter - there is only one for removal transaction
+      final String removedParticipant =
+          getRemovedParticipantFromParameter(privateTransaction.getPayload());
+
+      final PrivateTransactionEvent removalEvent =
+          new PrivateTransactionEvent(
+              privateTransaction.getPrivacyGroupId().get().toBase64String(), removedParticipant);
```

Comment:
Is there a chance for an NPE here?

Reply:
from _within_ the onchain precompile?

Reply:
looking at the code, the calling method returns earlier if privacy group id is empty. I can refactor to make that clearer?

Reply:
Unchecked `privateTransaction.getPrivacyGroupId().get()` ?

Reply:
You're right - I missed that in the diff. 

Any chance this function could be called from a different path?